desc/integrals/basis.py

```diff
@@ -58,6 +85,621 @@ def _in_epigraph_and(is_intersect, df_dy_sign, /):
     return put(is_intersect, idx[0], edge_case)
 
 
+def _chebcast(cheb, arr):
+    """Add leftmost axis to ``cheb`` depending on ``arr.ndim``.
+
+    Input ``arr`` should not have rightmost dimension of cheb that iterates
+    coefficients, but may have additional leftmost dimension for batch operation.
+    """
+    errorif(
+        jnp.ndim(arr) > cheb.ndim,
+        NotImplementedError,
+        msg=f"Only one additional axis for batch dimension is allowed. "
+        f"Got {jnp.ndim(arr) - cheb.ndim + 1} additional axes.",
+    )
+    return cheb if jnp.ndim(arr) < cheb.ndim else cheb[jnp.newaxis]
+
+
+class FourierChebyshevSeries(IOAble):
```

Comment:
in practice the Fourier part is what here? the "poloidal" (alpha) dependence? and the chebyshev is on zeta?

Reply:
Yes